wire-ios/Wire-iOS Tests/Backup/BackupLocalStoreMessagesTests.swift

```diff
@@ -0,0 +1,453 @@
+        XCTAssertEqual(result.insertionCount.successCount, 1000)
+        XCTAssertEqual(result.rehydrationCount.successCount, 1000)
+
+        try await context.perform { [fetchMessages] in
+            let fetchedMessages = try fetchMessages()
+            XCTAssertEqual(fetchedMessages.count, 1000)
+        }
+    }
+
+    // MARK: - Helper Methods
+
+    private func makeValidTextMessage(
+        conversationID: QualifiedID,
+        senderID: QualifiedID,
+        senderClientID: String? = "client-id"
+    ) -> MessageBackupModel {
+        MessageBackupModel(
+            id: UUID().uuidString.lowercased(),
+            conversationID: conversationID,
+            senderUserID: senderID,
+            senderClientID: senderClientID,
+            creationDate: Date(),
+            content: .text("Test message \(UUID().uuidString)")
+        )
+    }
+
+    private func makeValidAssetMessage(
+        conversationID: QualifiedID,
+        senderID: QualifiedID
+    ) -> MessageBackupModel {
+        MessageBackupModel(
+            id: UUID().uuidString.lowercased(),
+            conversationID: conversationID,
+            senderUserID: senderID,
+            senderClientID: "client-id",
+            creationDate: Date(),
+            content: .asset(
+                mimeType: "image/jpeg",
+                size: 1024,
+                name: "test.jpg",
+                otrKey: Data(repeating: 1, count: 32),
+                sha256: Data(repeating: 2, count: 32),
+                assetID: UUID().uuidString,
+                assetToken: "token",
+                assetDomain: "wire.com",
+                encryption: .aesGCM,
+                metadata: .image(width: 800, height: 600, tag: nil)
+            )
+        )
+    }
+
+    private func makeValidLocationMessage(
+        conversationID: QualifiedID,
+        senderID: QualifiedID
+    ) -> MessageBackupModel {
+        MessageBackupModel(
+            id: UUID().uuidString.lowercased(),
+            conversationID: conversationID,
+            senderUserID: senderID,
+            senderClientID: "client-id",
+            creationDate: Date(),
+            content: .location(
+                longitude: 8.5417,
+                latitude: 47.3769,
+                name: "Zurich",
+                zoom: 10
+            )
+        )
+    }
+
+    private func makeMessageWithInvalidNonce() -> MessageBackupModel {
+        MessageBackupModel(
+            id: "invalid-nonce-format",
+            conversationID: QualifiedID(id: UUID(), domain: "wire.com"),
+            senderUserID: QualifiedID(id: UUID(), domain: "wire.com"),
+            senderClientID: "client-id",
+            creationDate: Date(),
+            content: .text("This message has invalid nonce")
+        )
+    }
+
+    func fetchMessages() throws -> [ZMMessage] {
+        let fetchRequest = ZMMessage.fetchRequest()
+        let fetchResult = try context!.fetch(fetchRequest) as? [ZMMessage]
```

Comment:
```suggestion
        let fetchResult = try context.fetch(fetchRequest) as? [ZMMessage]
```

maybe rely on a computed var:
var context: NSManagedObjectContext {
coredataStack.syncContext
}

Reply:
[76e8aa5](https://github.com/wireapp/wire-ios/pull/3778/commits/76e8aa55f71da8ef0a8c913add2823935d779ee2)